### ISSUE TITLE
useFeature returns flag as `false` for one frame on page load, then `true`

### ISSUE THREAD

**Issue by bwiklund** [optimizely/react-sdk#58]
Everything about our setup seems pretty normal. This is what our code looks like in a FC, with unrelated code stripped out.

```
const [foo] = useFeature('foo');
console.log(foo);
```

Console output:
```
false
true
```

This is resulting in us rendering a big chunk of old UI for a single frame, then the new UI.

**Comment by bwiklund** [comment]
https://github.com/optimizely/react-sdk/blob/master/src/hooks.ts#L219

If i'm reading this hook correctly it looks like it will always return `isEnabled: false, isRead: true` on the first render, even if (like in our case) the parent component already knows our flag is `true` from an `<OptimizelyFeature>` component.

**Comment by bwiklund** [comment]
This `useEffect` has no chance to run before this hook returns on the first render, so there needs to be some synchronous code path that will set filters correctly on the first render if optimizely has already fetched the flags.

```
  useEffect(
    initializeWhenClientReadyFn(HookType.FEATURE, featureKey, optimizely, options, timeout, setState, getCurrentValues),
    [optimizely]
  );
```

**Comment by Danny-Driscoll** [comment]
Hey, thanks for raising this and sorry for the delayed response, we've been looking into and discussing this internally. We plan to pull a fix into an upcoming sprint, will provide update here as well when that's in.

**Comment by bwiklund** [comment]
any updates on this? this is a pretty painful bug

**Comment by mjc1283** [comment]
Hey @bwiklund, I am currently looking into a fix for this issue. Your ask to have the hook return up to date values on the first call makes sense.

In the meantime I wanted to let you know that `useFeature` returns two additional booleans representing the "ready" state of the client, and whether or not initialization timed out:
```js
const [enabled, variables, clientReady, didTimeout] = useFeature('my_feature');
console.log(enabled, clientReady, didTimeout);
```

 Our `OptimizelyFeature` component uses these booleans - when both are `false`, it renders nothing (see [here](https://github.com/optimizely/react-sdk/blob/6e6f1b43c00653e89da5a4dc8bc0fc7354bb8386/src/Feature.tsx#L46)). If you are happy with `OptimizelyFeature`'s behavior, you could implement the same check in your component that uses `useFeature`. Maybe this could help with the problem of rendering the old UI for one frame.

**Comment by mjc1283** [comment]
We released (v2.3.2)[https://github.com/optimizely/react-sdk/tree/2.3.2] which should resolve this issue.

**Comment by oMatej** [comment]
@mjc1283 I am not sure. I think I am facing a similar issue with SSR: #87 , it looks like even with provided `datafile` and `user` objects, Optimizely is not initialized immediately on the client side.  Sorry if the issue that I created is a duplicate of this one.

**Comment by pixelass** [comment]
We tried several methods and also get this issue. Using a datafile to initialize optimizely (as @oMatej mentioned) returns the same issue as using the provider by simply passing the `sdkKey`. This is a showstopper for us because features would always be disabled after SSR and then enabled on the client. @mjc1283 can you please take another look, why this might still be happening several minor versions later 🙏 . Your support is highly appreciated.

```
"@optimizely/react-sdk": "^2.6.3",
```

Sidenote. We ran several cases and did a lot of debugging. The optimizely instance is always available on the **server** and **client**, but the feature flags return `false` on the **server**, then `false` on the **client** (first render) and then `true` on the **client** (all following render cycles)

**Comment by yavorona** [comment]
@bwiklund @pixelass @oMatej thanks for following up on this issue and apologies for the late response. We just released an alpha patch version [2.7.1-alpha](https://github.com/optimizely/react-sdk/releases/tag/2.7.1-alpha) that hopefully resolves this issue. We would very much appreciate your feedback, so please feel free to provide us with your updates here. 

Thanks everyone for your patience as we continue working to resolve this issue!
